8-10-2020 update kristan

kristan-dev opened the pull request:
Updated controllers to have 3 classes

1. ProgramsController - deals with Program model. Uses @RestController and @ResponseStatus to build rest response.
2. StudentsController - deals with Student model. Uses @Controller and @ResponseEntity to build rest response.
3. Created SerializerService class to create abstraction for ObjectMapper methods. Uses @Component, @Autowired, @Qualifier
4. Updated Unit tests

Commit:
refactor applying dependency injection

Commit:
Created 2 controller classes with different implementation

StudentController implements @Controller, @ResponseEntity.
ProgramController implements @RestController, @ResponseStatus

Commit:
updated endpoints

jmcloud20 commented on kristan/src/main/java/com/example/springboot/controller/ProgramsController.java:
```diff
@@ -0,0 +1,107 @@
+        this.allPrograms = initProgramsRepo.getAllPrograms();
+    }
+
+    @RequestMapping(method = RequestMethod.GET, value = "", produces = "application/json")
+    @ResponseStatus(HttpStatus.OK)
+    public String allPrograms() {
+        String response = serializerService.convertObjectToString(this.allPrograms);
+        log.info(response);
+        return response;
+    }
```
Do not use @ResponseStatus when the return type is void. Use ResponseEntity instead.

kristan-dev replied:
Is this general advise, or particular the code snippet above? I asked, because the return type from the code snippet is String not void.

jmcloud20 commented on kristan/src/main/java/com/example/springboot/controller/ProgramsController.java:
```diff
@@ -0,0 +1,107 @@
+@Slf4j
+public class ProgramsController {
+
+    @Autowired
+    @Qualifier("serializer")
+    private SerializerService serializerService;
```
Instead of AutoWired, include serializer service in the constructor.

jmcloud20 commented on kristan/src/main/java/com/example/springboot/controller/ProgramsController.java:
```diff
@@ -0,0 +1,107 @@
+        this.allPrograms = initProgramsRepo.getAllPrograms();
+    }
+
+    @RequestMapping(method = RequestMethod.GET, value = "", produces = "application/json")
```
Use GetMapping for request method GET.

kristan-dev commented:
Hi @jmcloud20 , 

Thanks for the feedback. I've made the following updates...

1) Used ResponseEntity on all methods in ProgramsController class
2) Removed autowired and added SerializerService to the ProgramsController constructor
3) Used @GetMapping instead of RequestMapping for Get methods.

Commit:
Updated based on PR https://github.com/kristan-dev/upskilling/pull/13
